detection_rules/main.py

```diff
@@ -91,32 +98,96 @@ def generate_rules_index(ctx: click.Context, query, overwrite, save_files=True):
+    "-se",
+    type=click.Path(file_okay=False, exists=True),
+    help="Save imported exceptions to a directory",
+)
+def import_rules_into_repo(
+    input_file: click.Path,
+    required_only: bool,
+    exceptions_import: bool,
+    directory: click.Path,
+    save_directory: click.Path,
+    exceptions_directory: click.Path,
+):
     """Import rules from json, toml, yaml, or Kibana exported rule file(s)."""
-    rule_files = glob.glob(os.path.join(directory, '**', '*.*'), recursive=True) if directory else []
+    rule_files = glob.glob(os.path.join(directory, "**", "*.*"), recursive=True) if directory else []
     rule_files = sorted(set(rule_files + list(input_file)))
 
-    rule_contents = []
+    file_contents = []
     for rule_file in rule_files:
-        rule_contents.extend(load_rule_contents(Path(rule_file)))
+        file_contents.extend(load_rule_contents(Path(rule_file)))
+
+    if not file_contents:
+        click.echo("Must specify at least one file!")
+
+    exceptions_containers = {}
+    exceptions_items = {}
+
+    exceptions_containers, exceptions_items, _, unparsed_results = parse_exceptions_results_from_api(
+        file_contents, skip_errors=True
+    )
 
-    if not rule_contents:
-        click.echo('Must specify at least one file!')
+    file_contents = unparsed_results
 
-    for contents in rule_contents:
-        base_path = contents.get('name') or contents.get('rule', {}).get('name')
+    exception_list_rule_table = {}
+    for contents in file_contents:
+        base_path = contents.get("name") or contents.get("rule", {}).get("name")
         base_path = rulename_to_filename(base_path) if base_path else base_path
-        rule_path = os.path.join(RULES_DIR, base_path) if base_path else None
+        if base_path is None:
+            raise ValueError(f"Invalid rule file, please ensure the rule has a name field: {contents}")
+        rule_path = os.path.join(save_directory if save_directory is not None else RULES_DIRS[0], base_path)
 
         # handle both rule json formats loaded from kibana and toml
         data_view_id = contents.get("data_view_id") or contents.get("rule", {}).get("data_view_id")
         additional = ["index"] if not data_view_id else ["data_view_id"]
         rule_prompt(rule_path, required_only=required_only, save=True, verbose=True,
                     additional_required=additional, **contents)
+        if contents.get("exceptions_list"):
+
+            # For each item in rule.contents.data.exceptions_list to the exception_list_rule_table under the list_id
+            for exception in contents["exceptions_list"]:
+                exception_id = exception["list_id"]
+                if exception_id not in exception_list_rule_table:
+                    exception_list_rule_table[exception_id] = []
+                exception_list_rule_table[exception_id].append({"id": contents["id"], "name": contents["name"]})
+
+    # Build TOMLException Objects
+    if exceptions_import:
```

Comment:
nit: Same thing here, would be good to break this into an exception method since this cli command is getting long.

Reply:
++ will take a look after https://github.com/elastic/detection-rules/pull/3955 is merged.

Reply:
Done